### PR TITLE
Mtls + Keycloak 19 integration

### PR DESCRIPTION
**Pull request by pjgg**
### Summary

close: https://github.com/quarkus-qe/quarkus-test-suite/issues/768

Upgrade Keyckoak version to 18 or greater on `oidc-client-mutual-tls` scenario. 

OpenShift scenarios are going to use [RHSSO 7.6](https://github.com/quarkus-qe/quarkus-test-suite/blob/main/pom.xml#L647) which uses Keycloak 18 and is backward compatible with previous Keycloak configuration. 
Ref: https://access.redhat.com/articles/2342881

Baremetal is using Keycloak 19.0.1 due to a [flaky error](https://github.com/quarkus-qe/quarkus-test-suite/pull/782) on keycloak 18 on CI

UMA protection References:
* https://www.keycloak.org/docs/latest/authorization_services/index.html
* https://docs.kantarainitiative.org/uma/wg/oauth-uma-federated-authz-2.0-09.html

Please select the relevant options.

- [X] Refactoring

### Checklist:
- [X] Methods and classes used in PR scenarios are meaningful
- [X] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by michalvavrik** [comment]
FYI - I didn't review this as ATM it's the draft. Feel free to assign this to someone else as I'm OOTO next week.

File: security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/MutualTlsKeycloakService.java [hunk 1]
```diff
@@ -9,39 +8,28 @@
...
+    private final String realm;
+    private final String realmBasePath;
+    private final boolean openshiftScenario;
 
-    private MutualTlsKeycloakService(String realm) {
-        super(realm);
-        openshiftScenario = true;
-        this.realm = realm;
-    }
+    public static final String KC_DEFAULT_JKS_CMD_VALUE = "start-dev " +
```
Comment (fedinskiy) [inline security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/MutualTlsKeycloakService.java:17]: It would be nice to have some kind of explanation in the name. Since this variable is used only in context, such as this one: `@KeycloakContainer(command = KC_DEFAULT_JKS_CMD_VALUE,` I suggest to change the name to something akin to RUN_IN_DEVELOPMENT_MODE_JKS(I am not sure, that this command does, so the name is up to you)

File: security/oidc-client-mutual-tls/src/test/resources/keycloak-realm.json [hunk 2]
```diff
@@ -22,8 +22,14 @@
           "value": "test-normal-user"
         }
       ],
+      "clientRoles": {
+        "test-mutual-tls": [
+          "uma_protection"
```
Comment (fedinskiy) [inline security/oidc-client-mutual-tls/src/test/resources/keycloak-realm.json:27]: Why this change is needed?

**Review by fedinskiy — CHANGES_REQUESTED** [review]
We need a better explanation for what was done and why. I suggest to put it either in commit or PR description(or both)

File: security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/OpenShiftRhSsoOidcMtlsIT.java [hunk 3]
```diff
@@ -14,22 +17,47 @@
...
+            .withProperty("quarkus.oidc.client-id", CLIENT_ID_DEFAULT)
+            .withProperty("quarkus.oidc.tls.trust-store-file-type", JKS_KEYSTORE_FILE_EXTENSION)
+            .withProperty("quarkus.oidc.tls.key-store-file-type", JKS_KEYSTORE_FILE_EXTENSION)
+            .withProperty("quarkus.oidc.credentials.secret", CLIENT_SECRET_DEFAULT)
+            .withProperty("ks-file-extension", JKS_KEYSTORE_FILE_TYPE)
+            .withProperty("quarkus.oidc.tls.key-store-file", "rhsso-client-keystore.jks")
+            .withProperty("quarkus.oidc.tls.trust-store-file", "rhsso-client-truststore.jks")
+            .withProperty("ks-pwd", PASSWORD);
+
+    // createRestService(JKS_KEYSTORE_FILE_TYPE, JKS_KEYSTORE_FILE_EXTENSION, rhsso::getRealmUrl);
```
[the remark this answers is not in the document]
Comment (pjgg) [reply security/oidc-client-mutual-tls/src/test/java/io/quarkus/ts/security/oidcclient/mtls/OpenShiftRhSsoOidcMtlsIT.java]: leftover!

File: security/oidc-client-mutual-tls/src/test/resources/keycloak-realm.json (hunk 2 above)
Comment (pjgg) [reply security/oidc-client-mutual-tls/src/test/resources/keycloak-realm.json:27]: Since Keycloak 18 (maybe even in Keycloak 17), some server APIs are protected under `UMA_protection` role. 

https://www.keycloak.org/docs/latest/authorization_services/index.html

Comment (fedinskiy) [reply security/oidc-client-mutual-tls/src/test/resources/keycloak-realm.json:27]: Thank you, put this in the description, please

**Review by fedinskiy — APPROVED** [review]
